### ISSUE TITLE
1 ml of gasoline burns for 14 hours

### ISSUE THREAD

**Issue by aenodss** [CleverRaven/Cataclysm-DDA#73735]
### Describe the bug

While trying to learn about fire, I dropped a unit of gasoline and set fire to the tile containing it, it created a raging fire that lasted 14 hours.

### Attach save file

N/A

### Steps to reproduce

1. drop a unit of gasoline
2. set it on fire
3. wait

### Expected behavior

I expected that fire would last reasonably, similar to 0.G (about 7 minutes).

### Versions and configuration

- OS: windows
- Game version: cdda-experimental-2024-05-12-1705 39e2afa
- Graphics version: tiles
- Game language: System language
- Mods loaded: [[Dark Days Ahead [dda], Disable NPC Needs [no_npc_food], Portal Storms Ignore NPCs [personal_portal_storms], Slowdown Fungal Growth [no_fungal_growth]]

**Comment by XygenSS** [comment]
Confirmed

![Screenshot_2024-05-13-12-36-06-614_com cleverraven cataclysmdda experimental-edit](https://github.com/CleverRaven/Cataclysm-DDA/assets/27952407/aa80bb0b-d2e7-42b3-85c1-2c1032afe083)

- 1 unit of gasoline dropped on a brazier lasts for 24 hours
- the same dropped on the pavement instantly develops to a raging fire & seems to last just as long

- OS: Android
    - OS Version: Manufacturer: Xiaomi; Model: 23046RP50C; Release: 14; Incremental: V816.0.4.0.UMYCNXM; 
- Game Version: cdda-experimental-2024-05-01-1556 53a0a73 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

**Comment by TealcOneill** [comment]
/Confirm

- OS: Windows
    - OS Version: 10.0.19045.4291 (22H2)
- Game Version: cdda-experimental-2024-05-12-1705 39e2afa [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

**Comment by RenechCDDA** [comment]
I've bisected it to probably #73069 (yep, also me!). April 29: no issues, may 1st: issue. That PR was merged on the 29th and touched related code, so it is by far the most likely candidate.

...I am not finding the *reason* for this change in behavior, though.

**Comment by RenechCDDA** [comment]
Something I am not understanding is that the reported field age has become an incredibly large negative number.

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/84619419/eba9657f-829d-435b-a74c-76254c88fafc)

In versions before this issue appears, the field age is uniformly set to 601 right after lighting the fire.

**Comment by AnotherSeawhite** [comment]
Could this be related with #73495?

**Comment by RenechCDDA** [comment]
Okay. This is complicated. You can observe the change in behavior by adding the INCENDIARY ammo_effect back onto gasoline. Removing IGNITE is not sufficient - it must have the INCENDIARY tag. (This little fact tripped me up when initially testing)

Having INCENDIARY ammo_effect, at all causes the `gasoline` ammo to cookoff.

https://github.com/CleverRaven/Cataclysm-DDA/blob/731bdaad527d0c58abb34738878971cbb5dcdd0e/src/item_factory.cpp#L407-L408

Because it cooks off, it returns true in item::will_explode_in_fire(), causing the item to be erased when the fire field processes it (the processor assumes it exploded, **regardless of whether or not it did**) and does not pass it on to be considered a 'fuel'.

https://github.com/CleverRaven/Cataclysm-DDA/blob/731bdaad527d0c58abb34738878971cbb5dcdd0e/src/map_field.cpp#L972

https://github.com/CleverRaven/Cataclysm-DDA/blob/731bdaad527d0c58abb34738878971cbb5dcdd0e/src/map_field.cpp#L986

Because it is not considered a fuel, it does not simulate burn. Note that item::burn is passed fire_data as a reference, and it modifies that fire_data directly.

https://github.com/CleverRaven/Cataclysm-DDA/blob/731bdaad527d0c58abb34738878971cbb5dcdd0e/src/map_field.cpp#L991

But now that is not INCENDIARY and not "blowing up", it does get passed into item::simulate_burn() which has that previously mentioned behavior with charges. Also item::burn() itself is setting a huge negative number of charges (which itself should not be possible). There are many, many parts here that are working counterintuitively.

TL;DR: the game previously assumed it "blew up" and discounted it as fuel.

The simplest way to resolve this is probably to force IGNITE ammos to cookoff (diff below). This will restore the old behavior, but does not actually change the calculations which were giving crazy results. It merely goes back to skipping those calculations.

```diff --git a/src/item_factory.cpp b/src/item_factory.cpp
index 7c883c125b..5566d8e8d3 100644
--- a/src/item_factory.cpp
+++ b/src/item_factory.cpp
@@ -405,7 +405,8 @@ void Item_factory::finalize_pre( itype &obj )
             mats.find( material_oil ) == mats.end() ) {
             const auto &ammo_effects = obj.ammo->ammo_effects;
             obj.ammo->cookoff = ammo_effects.count( "INCENDIARY" ) > 0 ||
-                                ammo_effects.count( "COOKOFF" ) > 0;
+                                ammo_effects.count( "COOKOFF" ) > 0 ||
+                                ammo_effects.count( "IGNITE" ) > 0;
             static const std::set<std::string> special_cookoff_tags = {{
                     "SPECIAL_COOKOFF"
                 }
```
